src/common.rs

```diff
@@ -186,7 +186,7 @@ where
     );
 
     fn run(&mut self, (entities, errors, mut futures, mut pers): Self::SystemData) {
-        for (e, _) in (&*entities, &futures.check()).join() {
+        for (e, _) in (&*entities, &futures.mask().clone()).join() {
```

Comment:
This clone could potentially be removed if we had way to 'drain' storage while joining.

Reply:
Don't we have that? I just fixed the warning here (`check`) is deprecated.

Reply:
Yeah actually we have it.